### PR TITLE
Add headers to Hosting Version

### PR DESCRIPTION
**Pull request by rainshen49**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/20527

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
firebasehosting: added `headers` field in `google_firebase_hosting_version` resource (beta)
```

### CODE REVIEW

File: mmv1/products/firebasehosting/Version.yaml [hunk 1]
```diff
@@ -229,3 +236,34 @@ properties:
...
+                - 'glob'
+                - 'regex'
+            - name: 'regex'
+              type: String
+              description:
+                The user-supplied RE2 regular expression to match against the
+                request URL path.
+              exactly_one_of:
+                - 'glob'
+                - 'regex'
```
Comment (zli82016) [inline mmv1/products/firebasehosting/Version.yaml:264]: `exactly_one_of` doesn't work for the attributes inside the nested object of a list. It is a known issue. Can you please comment it out? https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#exactly_one_of

File: mmv1/products/firebasehosting/Version.yaml [hunk 2]
```diff
@@ -229,3 +236,34 @@ properties:
...
+          description: |
+            A Header specifies a URL pattern that, if matched to the request URL path, triggers Hosting
+            to apply the specified custom response headers.
+          type: NestedObject
+          properties:
+            - name: 'glob'
+              type: String
+              description:
+                The user-supplied glob to match against the request URL path.
+              exactly_one_of:
```
Comment (zli82016) [inline mmv1/products/firebasehosting/Version.yaml:254]: exactly_one_of doesn't work for the attributes inside the nested object of a list. It is a known issue. Can you please comment it out? https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#exactly_one_of

File: mmv1/products/firebasehosting/Version.yaml [hunk 3]
```diff
@@ -46,6 +46,13 @@ examples:
       site_id: 'site-id'
     test_env_vars:
       project_id: 'PROJECT_NAME'
+  - name: 'firebasehosting_version_headers'
```
Comment (zli82016) [inline mmv1/products/firebasehosting/Version.yaml:49]: Can you add another test example to cover the missing test for `regex` field?

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/products/firebasehosting/Version.yaml (hunk 3 above)
Comment (rainshen49) [reply mmv1/products/firebasehosting/Version.yaml:49]: Okay.

File: mmv1/products/firebasehosting/Version.yaml (hunk 2 above)
Comment (rainshen49) [reply mmv1/products/firebasehosting/Version.yaml]: Okay.

File: mmv1/products/firebasehosting/Version.yaml (hunk 1 above)
Comment (rainshen49) [reply mmv1/products/firebasehosting/Version.yaml]: Okay.

**Review by zli82016 — APPROVED** [review]
LGTM.

**Comment by zli82016** [comment]
@tylerg-dev , can you review this PR when you have time? Thanks.

**Review by tylerg-dev — APPROVED** [review]

**Pull request merged**